api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -1638,4 +1638,19 @@ public void checkIfPatientIdentifierTypesAreLocked() {
 	public List<PatientIdentifier> getPatientIdentifiersByPatientProgram(PatientProgram patientProgram) {
 		return dao.getPatientIdentifierByProgram(patientProgram);
 	}
+
+	/**
+	 * @see PatientService#getPatientsByIdentifier(org.openmrs.api.PatientService)
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Patient> getPatientsByIdentifier(String name, String identifier,
+			List<PatientIdentifierType> identifierTypes, boolean matchIdentifierExactly) throws APIException {
+		if (identifierTypes == null) {
+			identifierTypes = Collections.emptyList();
+		}
+
+		return Context.getPatientService().getPatients(name, identifier, identifierTypes, matchIdentifierExactly, 0, null);
```

Comment:
First of all, I do not see a new method added to the PatientDAO interface for the new method you added in HibernatePatientDAO.java  Secondly, this method should call that new method in the PatientDAO interface.  This is calling another service method that never uses the new method.